tempodb/backend/azure/azure_helpers.go

```diff
@@ -31,32 +32,40 @@ func GetContainerURL(ctx context.Context, conf *Config, hedge bool) (blob.Contai
 		retryOptions.TryTimeout = time.Until(deadline)
 	}
 
-	var httpSender pipeline.Factory
-	if hedge && conf.HedgeRequestsAt != 0 {
-		httpSender = pipeline.FactoryFunc(func(next pipeline.Policy, po *pipeline.PolicyOptions) pipeline.PolicyFunc {
-			return func(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
-				client := hedgedhttp.NewClient(conf.HedgeRequestsAt, uptoHedgedRequests, nil)
+	httpSender := pipeline.FactoryFunc(func(next pipeline.Policy, po *pipeline.PolicyOptions) pipeline.PolicyFunc {
+		return func(ctx context.Context, request pipeline.Request) (pipeline.Response, error) {
 
-				// Send the request over the network
-				resp, err := client.Do(request.WithContext(ctx))
+			customTransport := http.DefaultTransport.(*http.Transport).Clone()
 
-				return pipeline.NewHTTPResponse(resp), err
+			// add instrumentation
+			transport := newInstrumentedTransport(customTransport)
+
+			// hedge if desired (0 means disabled)
+			if hedge && cfg.HedgeRequestsAt != 0 {
+				transport = hedgedhttp.NewRoundTripper(cfg.HedgeRequestsAt, uptoHedgedRequests, transport)
 			}
-		})
-	}
+
+			client := http.Client{Transport: transport}
+
+			// Send the request over the network
+			resp, err := client.Do(request.WithContext(ctx))
```

Comment:
One final change. I was reviewing azure code and this function is called for every http request. Let's move lines 38->48 out of this function so we don't recreate the client on every single http request.

Thank you for this contribution!

Reply:
No problem! I had my suspicion as to whether was calling that function on every request but couldn't find it. That's been updated.